test/integration/instances_test.go

```diff
@@ -589,6 +589,211 @@ func TestInstance_withPG(t *testing.T) {
 	require.Equal(t, inst.PlacementGroup.PlacementGroupPolicy, pg.PlacementGroupPolicy)
 }
 
+func TestInstance_CreateWithReservedIPAddress(t *testing.T) {
+	client, teardown := createTestClient(t, "fixtures/TestInstance_CreateWithReservedIPAddress")
+	defer teardown()
+
+	// Reserve an IP for testing
+	reservedIP, err := client.ReserveIPAddress(context.Background(), linodego.ReserveIPOptions{Region: "us-east"})
+	if err != nil {
+		t.Fatalf("Failed to reserve IP: %v", err)
+	}
+	defer func() {
+		err := client.DeleteReservedIPAddress(context.Background(), reservedIP.Address)
+		if err != nil {
+			t.Errorf("Failed to delete reserved IP: %v", err)
+		}
+	}()
+
+	_, instanceTeardown, err := createInstanceWithReservedIP(t, client, reservedIP.Address)
+	if err != nil {
+		t.Fatalf("Error creating instance with reserved IP: %s", err)
+	}
+	defer instanceTeardown()
```

Comment:
This should be before the error check, so that the tear down still happens even when the `t.Fatal` is executed.

Reply:
Added defer statement immediately after the creation statement.